src/main/java/org/qiyana/model/LineItem.java

```diff
@@ -0,0 +1,59 @@
+package org.qiyana.model;
+
+import javax.persistence.*;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import java.util.Currency;
+
+@Entity
+@NoArgsConstructor
+@Getter
+public class LineItem {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  private Integer quantity;
+  private double totalItemPrice;
+  private Currency currency;
+
+  @ManyToOne
+  @JoinColumn(name = "order_id")
+  private Order order;
+
+  @OneToOne
+  @JoinColumn(name = "product_id", referencedColumnName = "id")
+  private Product product;
+
+  public LineItem(
+      Long id, Integer quantity,String currencyCode, Product product, Order order) {
+    this.id = id;
+    this.quantity = quantity;
+    this.product = product;
+    this.totalItemPrice = calculateTotalItemPrice();
+    this.currency = Currency.getInstance(currencyCode);
+    this.order = order;
+  }
+
+  private double calculateTotalItemPrice() {
```

Comment:
Great job. This is a clean and straightforward way of calculating the total price of the items. 

However, I suggest using `BigDecimal` instead of `double` here, as `double` can be quite imprecise when dealing with decimals. Read up on this here:
https://www.baeldung.com/java-double-vs-bigdecimal